urlcdab7d8364_jrwalsh_CycTools_tgz-pJ8-edu_iastate_cyctools_MainJ8/src/edu/iastate/cyctools/tools/load/view/LoadPanel.java

```diff
@@ -0,0 +1,1359 @@
+			else organization = ((Entry<String, String>)comboBoxOrganization.getSelectedItem()).getKey();
+			
+			batchEdits.createAuthorCredits(author, organization, controller.getConnection());
+		}
+	}
+	
+	private boolean searchFrameIDs() { //TODO convert to worker string task
+		JavacycConnection conn = controller.getConnection();
+		
+		TableModel tb = tableSpreadSheet.getModel();
+		
+		// Do a fast preliminary check to see if any user provided ID's are not frames in this database
+//		JOptionPane.showMessageDialog(this, "Validating import objects", "Validating", JOptionPane.PLAIN_MESSAGE);
+		int noMatchCount = 0;
+		for (int rowIndex = 0; rowIndex < tb.getRowCount(); rowIndex++) {
+			String userProvidedID = (String) tb.getValueAt(rowIndex, 0);
+			try {
+				if (!conn.frameExists(userProvidedID)) noMatchCount++;
+			} catch (PtoolsErrorException e) {
+				e.printStackTrace();
+			}
+		}
+		if (noMatchCount > 0) {
+			int n = JOptionPane.showConfirmDialog(this, "Some IDs do not match any objects in the database.  Would you like to perform a search for these objects in the database?"
+					, "Validating", JOptionPane.YES_NO_OPTION);
+			if (n == JOptionPane.NO_OPTION) return false;
+		}
+				
+		// For each user provided ID, do a search of that ID in the current DB.  Store results.
+		searchResults = new HashMap<String, ArrayList<Frame>>();
+		for (int rowIndex = 0; rowIndex < tb.getRowCount(); rowIndex++) {
+			String userProvidedID = (String) tb.getValueAt(rowIndex, 0);
+			try {
+				if (!searchResults.containsKey(userProvidedID)) {
+					searchResults.put(userProvidedID, conn.search(userProvidedID, ((Entry<String, String>)cmbImportType.getSelectedItem()).getKey()));
+				}
+			} catch (PtoolsErrorException e) {
+				e.printStackTrace();
+				searchResults.put(userProvidedID, null);
+			}
+		}
+		
+		// Copy the table model, but insert an extra column at the beginning to store the matched frame IDs
+		SortResults results = consumeSearchResults();
+		
+		tabbedSearchResults.setTitleAt(0, "Found " + results.getExactMatches().size() + " terms with FrameID matches in database");
+		tabbedSearchResults.setTitleAt(1, "Found " + results.getGoodMatches().size() + " terms with Synonym matches");
+		tabbedSearchResults.setTitleAt(2, (results.getAmbiguousMatches().size() + results.getNoMatches().size()) + " terms with ambiguous matches or no matches in database");
+		
+		labelSearchResults.setText("<html><body style='width: 100%'>We were able to match " + results.getExactMatches().size() + " out of " + searchResults.keySet().size() + " based on FrameIDs.  An " +
+				"additional " + results.getGoodMatches().size() + " out of " + searchResults.keySet().size() + " terms were matched based on synonyms.  This search was performed " +
+						"on " + ((Entry<String, String>)cmbImportType.getSelectedItem()).getKey() + " in the database " + controller.getSelectedOrganism() + ".");
+		
+		return true;
+	}
+	
+	private SortResults consumeSearchResults() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 30. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1074-1079 into a separate method.